core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -1922,9 +1922,8 @@ public DirectoryBrowserSupport doWs( StaplerRequest req, StaplerResponse rsp ) t
      */
     public HttpResponse doDoWipeOutWorkspace() throws IOException, ServletException, InterruptedException {
         checkPermission(Functions.isWipeOutPermissionEnabled() ? WIPEOUT : BUILD);
-        R b = getSomeBuildWithWorkspace();
-        FilePath ws = b!=null ? b.getWorkspace() : null;
-        if (ws!=null && getScm().processWorkspaceBeforeDeletion(this, ws, b.getBuiltOn())) {
```

Comment:
@jglick Let me try to come up with test case. w.r.t processWorkspaceBeforeDeletion, it's a deprecated method, we should not be making use of it. Please let me know if you have any other questions.